my-app/src/components/add.js

```diff
@@ -0,0 +1,45 @@
+import React, { Component } from 'react';
+
+class Add extends React.Component {
+  constructor (props) {
+      super(props);
+      this.state = {
+        items : this.props.items
+      }
+      this.addList = this.addList.bind(this);
+  }
+
+  componentWillReceiveProps(newProps) {
+    this.setState({
```

Comment:
Always check newProps verses oldProps, Otherwise it will do setState for same value